Develop/test docs

mpdesantis opened the pull request:
Just a bit of cleanup and brief READMEs in each directory. 
Sample program output added as a system test expectation.
Some minor printf modification just to unify the format between each subsystem's console output.
Did not format the printfs with newlines, since multithreaded console prints are hard to predict, did not know how to logically break apart. Maybe a TODO for later.

Thanks!

mpdesantis committed:
Brief READMEs for each test directory.

mpdesantis committed:
Unify format for console prints between subsystems.

mpdesantis committed:
Add system test notes with sample expected output.

orselane reviewed (commented):
Src changes: Print clarity, great

Test READMEs: should we separate it from the main readme? We do have a test section in there, but we should update it since we've changed the test folder quite a bit. If we keep it seperate, maybe just a single readme in the root test folder would be better to consolidate them? It just seems like a lot of documents to manage long term otherwise.

Expected output: I think we should keep a snippet, but describe what we're actually looking for. The output is not perfectly reproducible, but there is certain behavior we should be able to observe.

orselane commented on test/unit/README.md:
```diff
@@ -0,0 +1,9 @@
+Unit tests for relevant classes. 
+
+## Testing Notes
+* Only classes that expose public methods are unit tested.
```
This note is really important, I would maybe go ahead and add it as an extra assumption in main readme's scope section.

mpdesantis replied:
will do now good buddy

mpdesantis replied:
i've added this to main readme on separate branch and will push after this PR is closed, put it in the test section tho and listed that we're using JUnit 5.8.1 as test framework (that's the version we use, and the version we're supposed to use from Assignment00 doc).

orselane committed:
Main README.md test notes

orselane committed:
Adding expected output description

orselane committed:
Updated sample output

- Renamed to sample to differentiate with expected in case concurrency leads to a slightly different outcome

orselane committed:
Fixed elevator print times

orselane committed:
Updated sample-output with fixed floor nums

orselane committed:
Added floor number, improved lamp print.

orselane committed:
Added print to show destination sent.

orselane committed:
Setting to appropriate floor num to reach floor #7.

orselane committed:
Merge branch 'develop/latest' into develop/test-docs
